pages/doc.js

```diff
@@ -92,45 +112,47 @@ export default class Documentation extends Component {
       inputSelector: '#doc-search',
       debug: false // Set debug to true if you want to inspect the dropdown
     })
-  }
+  };
 
-  getLinkHref = (section, file) => {
-    const sectionSlug =
-      sidebar[section].slug || kebabCase(sidebar[section].name)
-    const fileSlug = file ? kebabCase(file.slice(0, -3)) : undefined
-    return `/doc/${compact([sectionSlug, fileSlug]).join('/')}`
-  }
+  getLinkHref = (section, file, subsection) => {
+    const sectionSlug = kebabCase(sidebar[section].name);
+    const subsectionSlug = subsection ? sidebar[section].files[subsection].indexFile.slice(0,-3) : undefined;
+    const fileSlug = file ? file.slice(0, -3) : undefined;
+    return `/doc/${compact([sectionSlug, subsectionSlug, fileSlug]).join('/')}`;
+  };
 
-  setCurrentPath = (section, file) => {
-    window.history.pushState(null, null, this.getLinkHref(section, file))
-  }
+  setCurrentPath = (section, file, subsection) => {
+    window.history.pushState(null, null, this.getLinkHref(section, file, subsection))
+  };
 
   onSectionSelect = (section, e) => {
-    e && e.preventDefault()
-    const { indexFile, files } = sidebar[section]
-    const file = indexFile || flatten(files)[0]
-    e && this.setCurrentPath(section, indexFile ? undefined : file)
-    this.loadFile({ file, section, parseHeadings: false })
-  }
-
-  onFileSelect = (file, section, e) => {
-    e && e.preventDefault()
-    this.setCurrentPath(section, file)
-    this.loadFile({ file, section, parseHeadings: true })
-  }
-
-  loadFile = ({ file, section, parseHeadings }) => {
-    fetch(`${sidebar[section].folder}/${file}`)
+    e && e.preventDefault();
+    const file = sidebar[section].hasOwnProperty('indexFile') ? sidebar[section]  : sidebar[section].files[0];
+    e && this.setCurrentPath(section);
+    this.loadFile({ file, section, parseHeadings: false });
+  };
+
+  onFileSelect = (file, section, e, subsection) => {
+    e && e.preventDefault();
+    this.setCurrentPath(section, file.indexFile, subsection);
+    this.loadFile({ file, section, subsection, parseHeadings: true });
+  };
+
+  loadFile = ({ file, section, subsection, parseHeadings }) => {
+    this.setState({load:true});
+    let folder = file.folder ? file.folder  : (subsection ? sidebar[section].files[subsection].folder : sidebar[section].folder);
+    fetch(`${folder}/${file.indexFile}`)
       .then(res => {
         res.text().then(text => {
           this.setState(
             {
               currentSection: section,
-              currentFile: file,
+              currentFile: file.folder ? `${file.folder}/${file.indexFile}` : (subsection ? `${sidebar[section].files[subsection].folder}/${file.indexFile}` : `${sidebar[section].folder}/${file.indexFile}`),
```

Comment:
too complicated, probably can be generalized or at least moved into a function that has some meaningful name

Reply:
fixed